Jenkinsfile.deploy

```diff
@@ -28,104 +41,86 @@ pipeline {
             ).trim()}"""
 
-        // Keep commented. It is for documentation only
-        // Cannot define here as an env variable defined here is immutable ...
-        // BRANCH_NAME => should be set by the multibranch pipeline. If single pipeline, then it is set with value from ${params.BUILD_BRANCH_NAME}
+        OPENSHIFT_API = credentials("OPENSHIFT_API")
+        OPENSHIFT_REGISTRY = credentials("OPENSHIFT_REGISTRY")
+        OPENSHIFT_INTERNAL_REGISTRY = "image-registry.openshift-image-registry.svc:5000"
+        // OPENSHIFT_CREDS => Credentials to access the Openshift cluster. Use in `loginOpenshift()`
 
-        // OPENSHIFT_API => Taken from Jenkins global env
-        // OPENSHIFT_REGISTRY => Taken from Jenkins global env
+        // Here under is a list of env variable which are defined dynamically into the pipeline
+        // We don't define them here because then there would be immutable ...
 
+        // Deploy image information
         // DEPLOY_IMAGE_REGISTRY_CREDENTIALS
         // DEPLOY_IMAGE_USE_OPENSHIFT
         // DEPLOY_IMAGE_REGISTRY
         // DEPLOY_IMAGE_NAMESPACE
         // DEPLOY_IMAGE_NAME_SUFFIX
         // DEPLOY_IMAGE_TAG
-
+        
         // TEST_PARAMS => parameters which will be set in `configure testing` stage, depending also on pipeline params
     }
     
     stages {
-        stage('Clean Workspace') {
-            steps {
-                dir ("${WORKING_DIR}") {
-                    deleteDir()
-                }
-            }
-        }
         stage('Initialize') {
             steps {
                 script {
-                    sh "mkdir -p ${WORKING_DIR} && cd ${WORKSPACE} && cp -Rap * ${WORKING_DIR}"
-                    sh "set +x && oc login --token=\$(oc whoami -t) --server=${OPENSHIFT_API} --insecure-skip-tls-verify"
+                    // Make sure Openshift is available and can authenticate before continuing
+                    loginOpenshift()
                     
                     setupImageEnvVarsFromParams("","DEPLOY")
 
                     if (params.DISPLAY_NAME != "") {
                         currentBuild.displayName = params.DISPLAY_NAME
                     }
 
-                    if (env.BRANCH_NAME != "") {
-                        // Switch to branch if not on a multibranch pipeline
-                        env.BRANCH_NAME = params.BUILD_BRANCH_NAME
-                        dir("${WORKING_DIR}"){
-                            checkout([$class: 'GitSCM', branches: [[name: env.BRANCH_NAME]], browser: [$class: 'GithubWeb', repoUrl: "${GIT_URL}"], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: '']], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'user-kie-ci10', url: "${GIT_URL}"]]])
-                        }
-                    }
+                    checkout(githubscm.resolveRepository("kogito-cloud-operator", params.GIT_AUTHOR, params.BUILD_BRANCH_NAME, false))
                 }
```

Comment:
Can't we use `githubscm.checkoutIfExists` function over here from shared library

Reply:
https://github.com/kiegroup/kogito-images/pull/182#discussion_r447714401